Formula/yosys.rb

```diff
@@ -0,0 +1,116 @@
+class Yosys < Formula
+  desc "Framework for Verilog RTL synthesis"
+  homepage "http://www.clifford.at/yosys/"
+  url "https://github.com/cliffordwolf/yosys/archive/yosys-0.7.tar.gz"
+  sha256 "3df986d0c6bf20b78193456e11c660f2ad935cc126537c2dc5726e78896d6e6e"
+
+  depends_on "python3"
+  depends_on "libffi" => :recommended
+  depends_on "readline" => :recommended
+  depends_on "pkg-config" => :build
+  depends_on "bison" => :build
+
+  # The makefile in Yosys 0.7 adds library search paths from macports, which a homebrew build
+  # should not be using. It also prints warnings about a missing brew command.
+  # See https://github.com/cliffordwolf/yosys/pull/303 for discussion.
+  # The patch is based on the Makefile changes in this upstream commit:
+  # https://github.com/cliffordwolf/yosys/commit/a431f4ee311b9563f546201d255e429e9ce58cfa
+  patch :DATA
```

Comment:
Does `https://github.com/cliffordwolf/yosys/commit/a431f4ee.patch` or `https://github.com/cliffordwolf/yosys/pull/303.patch` work at all? CC @ilovezfs for thoughts on whether this would be good to have in the formula-patches repo.